drivers/gpio/gpio_gecko.c

```diff
@@ -0,0 +1,425 @@
+	 *
+	 * Start with the GPIO module and set up the pin direction register.
+	 * 0 - pin is input, 1 - pin is output
+	 */
+
+	if (access_op == GPIO_ACCESS_BY_PIN) {
+		GPIO_PinModeSet(gpio_index, pin, mode, out);
+	} else {	/* GPIO_ACCESS_BY_PORT */
+		gpio_base->MODEL = (mode << 28) | (mode << 24) | (mode << 20)
+			| (mode << 16) | (mode << 12) | (mode << 8)
+			| (mode << 4) | mode;
+		gpio_base->MODEH = (mode << 28) | (mode << 24) | (mode << 20)
+			| (mode << 16) | (mode << 12) | (mode << 8)
+			| (mode << 4) | mode;
+		gpio_base->DOUT = (out ? 0xFFFF : 0x0000);
+	}
+
+	if (flags & GPIO_INT) {
+		enable_interrupt = true;
+		if (flags & GPIO_INT_EDGE) {
+			if (flags & GPIO_INT_ACTIVE_HIGH) {
+				interrupt_rising_edge = true;
+			} else if (flags & GPIO_INT_DOUBLE_EDGE) {
+				interrupt_rising_edge = true;
+				interrupt_falling_edge = true;
+			} else {
+				interrupt_falling_edge = true;
+			}
+		} else { /* GPIO_INT_LEVEL */
+			 /* should never get here, this is already handled */
+			return -ENOTSUP;
+		}
+	}
+
+	if (access_op == GPIO_ACCESS_BY_PIN) {
+		GPIO_IntConfig(gpio_index, pin, interrupt_rising_edge,
+			       interrupt_falling_edge, enable_interrupt);
+	} else {  /* GPIO_ACCESS_BY_PORT */
+		if (enable_interrupt) {
+			return -ENOTSUP;
+		}
+	}
+
+	return 0;
+}
+
+static int gpio_gecko_write(struct device *dev,
+			    int access_op, u32_t pin, u32_t value)
+{
+	const struct gpio_gecko_config *config = dev->config->config_info;
+	GPIO_P_TypeDef *gpio_base = config->gpio_base;
+
+	if (access_op == GPIO_ACCESS_BY_PIN) {
+		if (value) {
+			/* Set the data output for the corresponding pin.
+			 * Writing zeros to the other bits leaves the data
+			 * output unchanged for the other pins.
+			 */
+			gpio_base->DOUTSET = BIT(pin);
+		} else {
+			/* Clear the data output for the corresponding pin.
+			 * Writing zeros to the other bits leaves the data
+			 * output unchanged for the other pins.
+			 */
+			gpio_base->DOUTCLR = BIT(pin);
+		}
+	} else { /* GPIO_ACCESS_BY_PORT */
+		/* Write the data output for all the pins */
+		gpio_base->DOUT = value;
+	}
+
+	return 0;
+}
+
+static int gpio_gecko_read(struct device *dev,
+			   int access_op, u32_t pin, u32_t *value)
+{
+	const struct gpio_gecko_config *config = dev->config->config_info;
+	GPIO_P_TypeDef *gpio_base = config->gpio_base;
+
+	*value = gpio_base->DIN;
+
+	if (access_op == GPIO_ACCESS_BY_PIN) {
+		*value = (*value & BIT(pin)) >> pin;
+	}
+
+	/* nothing more to do for GPIO_ACCESS_BY_PORT */
+
+	return 0;
+}
+
+static int gpio_gecko_manage_callback(struct device *dev,
+				      struct gpio_callback *callback, bool set)
+{
+	struct gpio_gecko_data *data = dev->driver_data;
+
+	_gpio_manage_callback(&data->callbacks, callback, set);
+
+	return 0;
+}
+
+static int gpio_gecko_enable_callback(struct device *dev,
+				      int access_op, u32_t pin)
+{
+	struct gpio_gecko_data *data = dev->driver_data;
+
+	if (access_op == GPIO_ACCESS_BY_PIN) {
```

Comment:
I would have tested for ACCESS_BY_PORT here and return -ENOTSUP if so. No need of the else.

A bit like what you did with unsupported interrupt flags on the above functions: testing early for unsupported stuff, and leaving early then if so.

It's just a taste issue though.

Reply:
done

Reply:
done